src/gpu/SDL_gpu.c

```diff
@@ -586,6 +586,13 @@ SDL_GPUShaderFormat SDL_GetGPUShaderFormats(SDL_GPUDevice *device)
     return device->shader_formats;
 }
 
+SDL_PropertiesID SDL_GetGPUDeviceDebugProperties(SDL_GPUDevice *device)
+{
+    CHECK_DEVICE_MAGIC(device, 0);
+
+    return device->GetDeviceDebugProperties(device);
```

Comment:
We shouldn't have a separate function to get these, they should instead be a `props` member of SDL_GPUDevice that's allocated at creation time, similar to SDL_Renderer.

Reply:
I originally tried that but due to how device creation is done it would require a much larger change set to avoid code duplication in each backend. The device object is only created at the very end of the function so the property group object will need to be separately tracked throughout the create function and manually cleaned up on every failure return path.